src/software/new_geom/ray.h

```diff
@@ -0,0 +1,106 @@
+#pragma once
+
+#include "software/geom/line.h"
+#include "software/geom/point.h"
+
+class Ray final
+{
+   public:
+    /**
+     * Creates a degenerate Ray at (0, 0)
+     */
+    inline explicit constexpr Ray() {}
+
+    /**
+     * Creates a Ray with a start point and a direction angle
+     *
+     * @param start the start point of Ray
+     * @param direction the direction angle of Ray
+     */
+    inline explicit Ray(const Point& start, const Angle& direction)
+        : start_(start), direction_(direction)
+    {
+    }
+
+    /**
+     * Creates a Ray with a start point and a direction vector
+     *
+     * @param start the start point of Ray
+     * @param direction the direction vector of Ray
+     */
+    inline explicit Ray(const Point& start, const Vector& direction)
+        : Ray(start, direction.orientation())
+    {
+    }
+
+    /**
+     * Returns the start point of this Ray
```

Comment:
Just a thought. 

How would people feel about changing the 'start' of a Ray to the Origin?

Lemme know.

Reply:
Make the `start` of all `Ray`s the origin?

Reply:
Seems like a lot of use cases have `ball.position()` set as `start`, so it might need `start` to not always be at the origin? If it's a common use case to have `Ray` at the origin, I could just create an additional constructor that only takes an `Angle` and sets `start` to the origin.

Reply:
@EvanMorcom you don't just mean renaming `start` to `origin`? Otherwise we definitely have starting points of rays not being the origin at `(0, 0)`.

Reply:
I meant it as the 'reference origin' of the ray, not (0,0) in global coordinates.

Reply:
I would be fine with changing the `start` to `origin`

Reply:
I think naming things `origin` is a bit confusion for the reason Mat was confused above.

Reply:
I was mostly confused because the way evan wrote his comment sounded like a functional change rather than just a naming change. I think in either case it's pretty obvious that `some_ray.start()` vs `some_ray.origin()` indicates the starting point of the ray, but I'm leaning towards leaving `start` because I can see how this could be confusing.